src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -256,8 +279,6 @@ public override int Read(byte[] array, int offset, int count)
                     break;
                 }
 
-                Debug.Assert(_inflater.NeedsInput(), "We can only run into this case if we are short of input");
```

Comment:
This debug was causing exceptions because NeedsInput() was calling the handle that was at that time uninitialized. I considered adding a check around the handle, but determined the slowdown not worth the sake of this debug.